src/components/icon/Icon.js

```diff
@@ -1,41 +1,53 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import styles from './Icon.css';
 
-const SpriteIcon = ({ svg, ...rest }) => (
-    <i { ...rest }>
-        <svg viewBox={ svg.viewBox }>
-            <use xlinkHref={ svg.symbol } />
-        </svg>
-    </i>
-);
-
-SpriteIcon.propTypes = {
-    svg: PropTypes.object.isRequired,
-};
-
-const InlineIcon = ({ svg, ...rest }) => <i { ...rest } dangerouslySetInnerHTML={ { __html: svg } } />;
-
-InlineIcon.propTypes = {
-    svg: PropTypes.string.isRequired,
-};
-
-const Icon = (props) => {
-    const { svg, className, ...rest } = props;
-    const finalProps = {
-        ...rest,
-        svg,
-        className: classNames(styles.icon, className),
-    };
-
-    return typeof svg === 'string' ?
-        <InlineIcon { ...finalProps } /> :
-        <SpriteIcon svg={ svg } { ...finalProps } />;
-};
+class Icon extends Component {
+    constructor(props) {
+        super(props);
+
+        if (typeof props.svg === 'string') {
+            this.state = {
+                contents: props.svg,
+            };
+        } else {
+            this.state = {
+                contents: null,
+            };
```

Comment:
I would change this to:
```js
this.state = {
    contents: typeof props.svg === 'string' ? props.svg : null
}
```

or 

```js
const contents = typeof props.svg === 'string' ? props.svg : null;

this.state = { contents };
```

Reply:
@paulobmarcos suggested the same thing. I changed it to your first alternative 👍